base/loading.jl

```diff
@@ -367,15 +368,15 @@ function locate_package(pkg::PkgId)::Union{Nothing,String}
         for env in load_path()
             # look for the toplevel pkg `pkg.name` in this entry
             found = project_deps_get(env, pkg.name)
-            found === nothing && continue
-            if pkg == found
-                # pkg.name is present in this directory or project file,
-                # return the path the entry point for the code, if it could be found
-                # otherwise, signal failure
-                return implicit_manifest_uuid_path(env, pkg)
+            if found !== nothing
+                @assert found.name == pkg.name
+                if found.uuid === nothing
+                    # pkg.name is present in this directory or project file,
+                    # return the path the entry point for the code, if it could be found
+                    # otherwise, signal failure
+                    return implicit_manifest_uuid_path(env, pkg)
+                end
             end
-            @assert found.uuid !== nothing
-            return locate_package(found) # restart search now that we know the uuid for pkg
```

Comment:
Can `locate_package(found)` be added back in? As is, this change results in `Base.locate_package(Base.PkgId("Pkg"))` returning `nothing`:

```julia
❯ ~/julia/julia --startup-file=no
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.9.0-DEV.893 (2022-07-05)
 _/ |\__'_|_|_|\__'_|  |  Commit 8a776bda4c (0 days old master)
|__/                   |

julia> Base.locate_package(Base.PkgId("Pkg"))

julia>

```

Vs.

```julia
❯ julia --startup-file=no
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.3 (2022-05-06)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> Base.locate_package(Base.PkgId("Pkg"))
"/home/awadell/.local/opt/spack/opt/spack/linux-opensuse_leap15-zen3/gcc-11.2.0/julia-1.7.3-fkmbucihyf3jb3jx7plzg6eoqglmh7uz/share/julia/stdlib/v1.7/Pkg/src/Pkg.jl"

julia>

```

Or is the correct move to use `Base.locate_package(Base.identify_package("Pkg"))`?

Reply:
Neither is "correct" In that both are ambiguous about what they are supposed to return. There is `find_package("Pkg")` for "internal-Pkg-use" (according to the comment), and other places are usually expected to load a Manifest.toml and locate packages by their `[deps]` section